pkg/dockerregistry/server/simplelookupicsp.go

```diff
@@ -41,7 +47,18 @@ func (s *simpleLookupICSP) FirstRequest(
 		return []reference.DockerImageReference{ref.AsRepository()}, nil
 	}
 
-	imageRefList, err := s.alternativeImageSources(ref, icspList.Items)
+	idmsList, err := s.idmsClient.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("unable to list IDMS config: %s", err)
+		return []reference.DockerImageReference{ref.AsRepository()}, nil
+	}
+
+	if len(icspList.Items) > 0 && len(idmsList.Items) > 0 {
+		klog.Errorf("error finding both icsp and idms resources at the same time: %s", err)
+		return []reference.DockerImageReference{ref.AsRepository()}, nil
```

Comment:
I think we should surface an error here so that this doesn't go unnoticed to the caller.

Reply:
Addressed this in line68 and line73.